lib/report_portal/cucumber/report.rb

```diff
@@ -183,24 +184,46 @@ def start_feature_with_parentage(feature, desired_time)
               tags = []
               type = :SUITE
             else
+              # TODO: Consider adding feature description and comments.
               name = "#{feature.keyword}: #{feature.name}"
-              description = feature.file # TODO: consider adding feature description and comments
+              description = feature.file
               tags = feature.tags.map(&:name)
               type = :TEST
             end
-            # TODO: multithreading # Parallel formatter always executes scenarios inside the same feature in the same process
-            if parallel? &&
-                index < path_components.size - 1 && # is folder?
-                (id_of_created_item = ReportPortal.item_id_of(name, parent_node)) # get id for folder from report portal
-              # get child id from other process
+            is_created = false
+            if parallel? && name.include?("Folder:")
+              folder_name = name.gsub("Folder: ", "")
+              $folder_creation_tracking_file = ".reportportal/folder_creation_tracking_#{ReportPortal.launch_id}.lck"
+              File.open($folder_creation_tracking_file, 'r+') do |f|
+                f.flock(File::LOCK_SH)
+                report_portal_folders = f.read
+                if report_portal_folders
+                  report_portal_folders_array = report_portal_folders.split(/\n/)
+                  if report_portal_folders_array.include?(folder_name)
```

Comment:
will we have issue with following solution structure?
tests
-folder1
--the_same_name_folder
-folder2
--the_same_name_folder

Reply:
I will look into this

Reply:
We might, I will have to look into that more closely, but will make the easy fixes now.

Reply:
I think I resolved this issue. I will comment on the logic that handles this.